modules/post/windows/gather/enum_shares.rb

```diff
@@ -12,174 +12,199 @@ def initialize(info = {})
   def enum_run_unc(base_key)
-    unc_paths = []
-    full_path = base_key + "\\Software\\Microsoft\\Windows\\CurrentVersion\\Explorer\\RunMRU"
+    full_path = base_key + '\\Software\\Microsoft\\Windows\\CurrentVersion\\Explorer\\RunMRU'
     vals_found = registry_enumvals(full_path)
-    if vals_found
-      vals_found.each do |k|
-        if k =~ /./
-          run_entrie = registry_getvaldata(full_path, k)
-          unc_paths << run_entrie if run_entrie =~ /^\\\\/
-        end
-      end
+
+    return [] unless vals_found
+
+    unc_paths = []
+    vals_found.each do |k|
+      next if k.include?('MRUList')
+
+      run_entry = registry_getvaldata(full_path, k).to_s
+      unc_paths << run_entry.gsub(/\\1$/, '') if run_entry.starts_with?('\\\\')
     end
 
-    return unc_paths
+    unc_paths
   end
 
   # Method for enumerating configured shares on a target box
-  def enum_conf_shares()
-    if sysinfo["OS"] =~ /Windows 7|Vista|2008/
-      shares_key = "HKEY_LOCAL_MACHINE\\SYSTEM\\CurrentControlSet\\services\\LanmanServer\\Shares"
-    else
-      shares_key = "HKEY_LOCAL_MACHINE\\SYSTEM\\CurrentControlSet\\services\\lanmanserver\\Shares"
+  #
+  # @return [Array] List of network shares in the form of [ name, type, remark, path ]
+  def enum_conf_shares
+    shares_key = nil
+
+    [
+      'HKEY_LOCAL_MACHINE\\SYSTEM\\CurrentControlSet\\services\\LanmanServer\\Shares',
+      'HKEY_LOCAL_MACHINE\\SYSTEM\\CurrentControlSet\\services\\lanmanserver\\Shares'
+    ].each do |k|
+      if registry_key_exist?(k)
+        shares_key = k
+        break
+      end
+    end
+
+    if shares_key.blank?
+      print_status('No network shares were found')
+      return
     end
+
     share_names = registry_enumvals(shares_key)
 
-    if share_names.length > 0
-      shares = []
-      print_status("The following shares were found:")
-      share_names.each do |sname|
-        info = registry_getvaldata(shares_key, sname)
-        next if info.nil?
-
-        share_info = info.split("\000")
-        print_status("\tName: #{sname}")
-        stype = remark = path = nil
-        share_info.each do |e|
-          name, val = e.split("=")
-          case name
-          when "Path"
-            print_status "\tPath: #{val}"
-            path = val
-          when "Type"
-            print_status "\tType: #{val}"
-            stype = share_type(val.to_i)
-          when "Remark"
-            remark = val
-          end
+    if share_names.empty?
+      print_status('No network shares were found')
+      return
+    end
+
+    shares = []
+    print_status('The following shares were found:')
+    share_names.each do |sname|
+      info = registry_getvaldata(shares_key, sname)
+      next if info.nil?
+
+      print_status("\tName: #{sname}")
+
+      # NOTE: Extracting these values will fail on Meterpreter.
+      # Share info keys are REG_MULTI_SZ. Meterpreter stdapi registry
+      # query_value() function returns malformed REG_MULTI_SZ values.
```

Comment:
This was broken both before and after this PR (https://github.com/rapid7/metasploit-payloads/issues/582).

```
#<Rex::Post::Meterpreter::Extensions::Stdapi::Sys::RegistrySubsystem::RegistryValue:0x000055e45e5363b0 @client=#<Session:meterpreter 192.168.200.218:51781 (192.168.200.218) "CORP\user @ WIN-17B09RRRJTG">, @hkey=0, @name="NETLOGON", @type=7, @data="C\x00S\x00C\x00F\x00l\x00a\x00g\x00s\x00=\x002\x005\x006\x00\x00\x00M\x00a\x00x\x00U\x00s\x00e\x00s\x00=\x004\x002\x009\x004\x009\x006\x007\x002\x009\x005\x00\x00\x00P\x00a\x00t\x00h\x00=\x00C\x00:\x00\\\x00W\x00i\x00n\x00d\x00o\x00w\x00s\x00\\\x00S\x00Y\x00S\x00V\x00O\x00L\x00\\\x00s\x00y\x00s\x00v\x00o\x00l\x00\\\x00c\x00o\x00r\x00p\x00.\x00l\x00o\x00c\x00a\x00l\x00\\\x00S\x00C\x00R\x00I\x00P\x00T\x00S\x00\x00\x00P\x00e\x00r\x00m\x00i\x00s\x00s\x00i\x00o\x00n\x00s\x00=\x000\x00\x00\x00R\x00e\x00m\x00a\x00r\x00k\x00=\x00L\x00o\x00g\x00o\x00n\x00 \x00s\x00e\x00r\x00v\x00e\x00r\x00 \x00s\x00h\x00a\x00r\x00e\x00 \x00\x00\x00S\x00h\x00a\x00r\x00e\x00N\x00a\x00m\x00e\x00=\x00N\x00E\x00T\x00L\x00O\x00G\x00O\x00N\x00\x00\x00T\x00y\x00p\x00e\x00=\x000\x00\x00\x00\x00\x00">
```

I believe Op3n4M3 may be looking into this, but should not be considered a blocker for this PR.